File: _archive/api-py/app/models/notification.py
import uuid
from datetime import datetime, timezone
from sqlalchemy import String, Boolean, DateTime, Enum as SAEnum, ForeignKey, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship
from sqlalchemy.dialects.postgresql import UUID, JSONB
from app.database import Base
from app.models.enums import NotificationType


class Notification(Base):
    __tablename__ = 'notifications'

    id: Mapped[str] = mapped_column(UUID(as_uuid=False), primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id: Mapped[str] = mapped_column(UUID(as_uuid=False), ForeignKey('users.id', ondelete='CASCADE'), index=True)
    notif_type: Mapped[NotificationType] = mapped_column('type', SAEnum(NotificationType))
    title: Mapped[str] = mapped_column(String(200))
    title_ar: Mapped[str | None] = mapped_column(String(200), nullable=True)
    body: Mapped[str | None] = mapped_column(Text, nullable=True)
    body_ar: Mapped[str | None] = mapped_column(Text, nullable=True)
    data: Mapped[dict | None] = mapped_column(JSONB, nullable=True)
    is_read: Mapped[bool] = mapped_column(Boolean, default=False, index=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc), index=True)

    user = relationship('User', back_populates='notifications')

File: packages/shared/types/notification.ts
export enum NotificationType {
  ORDER_UPDATE = 'ORDER_UPDATE',
  PAYMENT_RECEIVED = 'PAYMENT_RECEIVED',
  DELIVERY_UPDATE = 'DELIVERY_UPDATE',
  CONFIRMATION_REQUIRED = 'CONFIRMATION_REQUIRED',
  CONFIRMATION_APPROVED = 'CONFIRMATION_APPROVED',
  PROMOTIONAL = 'PROMOTIONAL',
  SYSTEM_ALERT = 'SYSTEM_ALERT',
  REVIEW_REMINDER = 'REVIEW_REMINDER',
  NEW_MESSAGE = 'NEW_MESSAGE',
  STAFF_ASSIGNED = 'STAFF_ASSIGNED',
  LOW_STOCK_ALERT = 'LOW_STOCK_ALERT',
}

export interface Notification {
  id: string;
  userId: string;
  type: NotificationType;
  title: string;
  titleAr: string;
  body: string;
  bodyAr: string;
  data?: Record<string, unknown>;
  isRead: boolean;
  readAt?: string;
  image?: string;
  actionUrl?: string;
  createdAt: string;
}
